src/sentry/api/endpoints/release_thresholds/release_threshold_status_index.py

```diff
@@ -126,61 +161,161 @@ def get(self, request: Request, organization: Organization | RpcOrganization) ->
                 for threshold in thresholds_list:
-                    is_healthy = self.is_threshold_healthy(threshold)
-                    project_threshold_statuses.append(
-                        {
-                            **serialize(threshold),
-                            "is_healthy": is_healthy,
+                    if threshold.threshold_type not in thresholds_by_type:
+                        thresholds_by_type[threshold.threshold_type] = {
+                            "projects": [],
+                            "releases": [],
+                            "thresholds": [],
                         }
+                    thresholds_by_type[threshold.threshold_type]["projects"].append(project.id)
+                    thresholds_by_type[threshold.threshold_type]["releases"].append(release.version)
+                    enriched_threshold: EnrichedThreshold = {
+                        # see https://github.com/python/mypy/issues/4122
+                        **serialize(threshold),  # type: ignore
+                        "key": self.construct_threshold_key(
+                            release=release, project=project, threshold=threshold
+                        ),
+                        "start": release.date,  # deploy.date_finished _would_ be more accurate, but is not keyed on project so cannot be used
+                        "end": release.date
+                        + timedelta(threshold.window_in_seconds),  # start + threshold.window
+                        "release": release.version,
+                        "project_id": project.id,
+                        "is_healthy": False,
+                    }
+                    thresholds_by_type[threshold.threshold_type]["thresholds"].append(
+                        enriched_threshold
                     )
-                release_project[project.id] = project_threshold_statuses
-            release_threshold_health[release.id] = release_project
+
+        # ========================================================================
+        # Step 4: Determine threshold status per threshold type and return results
+        # ========================================================================
+        release_threshold_health = defaultdict(list)
+        for threshold_type, filter_list in thresholds_by_type.items():
+            project_id_list = [proj_id for proj_id in filter_list["projects"]]
+            release_value_list = [release_version for release_version in filter_list["releases"]]
+            category_thresholds: List[EnrichedThreshold] = filter_list["thresholds"]
+            if threshold_type == ReleaseThresholdType.TOTAL_ERROR_COUNT:
+                """
+                Fetch errors timeseries for all projects with an error_count threshold in desired releases
+                Iterate through timeseries given threshold window and determine health status
+
+                TODO: determine whether results have been truncated or not
```

Comment:
Are these TODOs to be done in future or like the next PR?

Reply:
This comment can be removed.

The TLDR here is that Snuba only returns up to a maximum of 10k results (default 1k).
this means we may not have a full set of data to make a proper determination of health.
The conclusion decided upon was that if we determine the window is too large and we cannot guarantee accurate results, then we will throw an error and request the customer narrow their search window